docs/www/tls-kubernetes.md

```diff
@@ -0,0 +1,375 @@
+
+For example, let's communicate with our cluster using `rpk`.
+
+Depending on the the connectivity, first we need to get a list of brokers.
+
+For convenience, let's save the result into a variable called `$BROKERS`:
+
+Internal:
+
+```bash
+export BROKERS=kubectl get clusters cluster -o=jsonpath='{.status.nodes.internal}' | jq -r 'join(",")'
+```
+
+External: 
+
+```bash
+export BROKERS=kubectl get clusters cluster -o=jsonpath='{.status.nodes.external}' | jq -r 'join(",")'
+```
+
+After that we can run our `rpk` commands. 
+
+For example if we only have TLS enabled, we just need to provide the CA: 
+
+```
+kubectl exec cluster-sample-tls-0 -- \
+rpk cluster info  --brokers $BROKERS --tls-truststore=/etc/tls/certs/ca.crt 
+```
+
+For mTLS, we need to provide both the CA and the key-pair:
+
+```
+kubectl exec cluster-sample-tls-0 -- \
+rpk cluster info  --brokers $BROKERS --tls-truststore=/etc/tls/certs/ca.crt --tls-key=/etc/tls/certs/tls.key --tls-cert=/etc/tls/certs/tls.crt
+```
+
+Typically, a client runs in a separate pod. If that pod is within the same Kubernetes cluster and namespace, the client pod can directly mount the certificate secrets:
+
+```
+spec:
+  template:
+    spec:
+      volumes:
+        - name: _tlscert_
+          secret:
+            secretName: <cluster>-redpanda
+...
+ containers:
+        - name: client
+          volumeMounts:
+            - mountPath: /etc/tls/certs
+              name: _tlscert_
+```
+
+You can find a working example with mTLS here: [https://github.com/vectorizedio/redpanda/tree/dev/src/go/k8s/tests/e2e/create-topic-with-client-auth](https://github.com/vectorizedio/redpanda/tree/dev/src/go/k8s/tests/e2e/create-topic-with-client-auth)
+
+If the client pod runs in a separate namespace within the same Kubernetes cluster, the secret needs to be copied over.
+
+Similarly, if the client is on a separate network, but accessible through an external IP address or name (see [https://vectorized.io/docs/kubernetes-connectivity](https://vectorized.io/docs/kubernetes-connectivity)) the secrets need to become available by copying over the resource or exporting the CA `ca.crt` and key-pair `tls.crt` and `tls.key` as files. For example, to retrieve tls.crt:
+
+```
+kubectl get secret <cluster>-user-client -o go-template='{{index .data "ca.crt"}}' | base64 -d - > tls.crt
+```
+
+Alternatively, you can retrieve the whole resource:
+
+```
+kubectl get secret <cluster>-_user-client_ --namespace=default -o yaml
+```
+
+Once the secrets are available as files, it is possible to use `rpk` to connect with the brokers. Here we consider external brokers, but they could be replaced with internal addresses if accessible:
+
+For convenience, let's create a config file and store our values there. You can pass this as arguments to your `rpk` commands but they can become very long.
```

Comment:
```suggestion
For this example, let's create a configuration file and store our values there. We'll create `rpk-config.yaml`:
```

Reply:
```For convenience, let's create a file and store our values there. We can pass this whole file as an argument for `rpk` instead of having to retype it all over again as flags. Let's name it `rpk-config.yaml`:
```
@coral-waters wdyt?

Reply:
Awesome